enumerable_methods.rb

```diff
@@ -0,0 +1,182 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    instance = self
+    instance = instance.clone
+    return instance.to_enum unless block_given?
+
+    i = 0
+    loop do
+      yield(instance[i])
+      i += 1
+      break if i == instance.length
+    end
+  end
+
+  def my_each_with_index
+    instance = self
+    instance = instance.clone
+    return instance.to_enum unless block_given?
+
+    i = 0
+    loop do
+      yield(instance[i], i)
+      i += 1
+      break if i == instance.length
+    end
+  end
+
+  def my_select
+    instance = self
+    instance = instance.clone
+    result = []
+    return instance.to_enum unless block_given?
+
+    instance.my_each do |e|
+      result.push(e) if yield(e)
+    end
+    result
+  end
+
+  def my_all?(type = nil)
+    instance = self
+    instance = instance.clone
+    puts 'warning: given block not used' if block_given? && !type.nil?
+    instance.my_each do |e|
+      is_true = truthy(e, type)
+      is_true ||= truthy_no_typeblock(e, type.nil?, !block_given?)
+      return false unless is_true
+
+      if block_given?
+        return false unless yield(e)
+      end
+    end
+    true
+  end
+
+  def my_any?(type = nil)
+    instance = self
+    instance = instance.clone
+    puts 'warning: given block not used' if block_given? && !type.nil?
+    !instance.my_none?(type)
+  end
+
+  def my_none?(type = nil)
+    instance = self
+    instance = instance.clone
+    puts 'warning: given block not used' if block_given? && !type.nil?
+    instance.my_each do |e|
+      is_true = truthy(e, type)
+      is_true ||= truthy_no_typeblock(e, type.nil?, !block_given?)
+      return false if is_true
+      return false if block_given? && yield(e)
+    end
+    true
+  end
```

Comment:
`p [].my_none?` returns `false` instead of `true`
`p [nil].my_none?` returns `false` instead of `true`
`p [nil, false].my_none?` returns `false` instead of `true`